core/domain/user_jobs_continuous.py

```diff
@@ -428,15 +428,29 @@ def reduce(key, stringified_values):
         # Find the average of all average ratings
         ratings = [value['average_rating_for_owned_exp'] for value in values
                    if value.get('average_rating_for_owned_exp')]
+        mr_model = user_models.UserStatsModel.get(key, strict=False)
+
         if len(ratings) != 0:
             average_ratings = (sum(ratings) / float(len(ratings)))
-            user_models.UserStatsModel(
-                id=key,
-                impact_score=user_impact_score,
-                total_plays=total_plays,
-                average_ratings=average_ratings).put()
+            is_average_ratings = True
+        else:
+            is_average_ratings = False
```

Comment:
(This thought had crossed my mind during writing this code. But I went for creating a separate flag for this instead)
Done.